packages/@react-spectrum/slider/src/RangeSlider.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import {SliderBase, SliderBaseChildArguments, SliderBaseProps} from './SliderBase';
+import {SpectrumRangeSliderProps} from '@react-types/slider';
+import styles from '@adobe/spectrum-css-temp/components/slider/vars.css';
+import {useHover} from '@react-aria/interactions';
+import {useLocale} from '@react-aria/i18n';
+import {VisuallyHidden} from '@react-aria/visually-hidden';
+
+function RangeSlider(props: SpectrumRangeSliderProps) {
+  let {onChange, value, defaultValue, ...otherProps} = props;
+
+  let baseProps: Omit<SliderBaseProps, 'children'> = {
+    ...otherProps,
+    count: 2,
+    value: value != null ? [value.start, value.end] : undefined,
+    defaultValue: defaultValue != null ? [defaultValue.start, defaultValue.end] :
+      // make sure that useSliderState knows we have two handles
+      [props.minValue ?? DEFAULT_MIN_VALUE, props.maxValue ?? DEFAULT_MAX_VALUE],
+    onChange(v) {
+      onChange?.({start: v[0], end: v[1]});
+    }
+  };
+
+  let {direction} = useLocale();
+  let hovers = [useHover({}), useHover({})];
+
+  return (
+    <SliderBase {...baseProps} classes={'spectrum-Slider--range'}>
+      {({state, thumbProps, inputRefs, inputProps, ticks}: SliderBaseChildArguments) => {
+        let cssDirection = direction === 'rtl' ? 'right' : 'left';
+
+        let lowerTrack = (<div
+          className={classNames(styles, 'spectrum-Slider-track')}
+          style={{width: `${state.getThumbPercent(0) * 100}%`}} />);
+        let middleTrack = (<div
+          className={classNames(styles, 'spectrum-Slider-track')}
+          style={{[cssDirection]: `${state.getThumbPercent(0) * 100}%`, width: `${Math.abs(state.getThumbPercent(0) - state.getThumbPercent(1)) * 100}%`}} />);
+        let upperTrack = (<div
+          className={classNames(styles, 'spectrum-Slider-track')}
+          style={{[cssDirection]: `${state.getThumbPercent(1) * 100}%`, width: `${(1 - state.getThumbPercent(1)) * 100}%`}} />);
+
+        let handles = [0, 1].map(i => (<div
+          className={classNames(styles, 'spectrum-Slider-handle', {'is-hovered': hovers[i].isHovered, 'is-dragged': state.isThumbDragging(i)})}
+          style={{[cssDirection]: `${state.getThumbPercent(i) * 100}%`}}
+          {...mergeProps(thumbProps[i], hovers[i].hoverProps)}
+          role="presentation">
+          <VisuallyHidden isFocusable>
+            <input className={classNames(styles, 'spectrum-Slider-input')} ref={inputRefs[i]} {...inputProps[i]} />
+          </VisuallyHidden>
+        </div>));
+
+        return (<>
+          {lowerTrack}
+          {ticks}
+          <FocusRing within focusRingClass={classNames(styles, 'is-focused')}>
+            {handles[0]}
+          </FocusRing>
+          {middleTrack}
+          <FocusRing within focusRingClass={classNames(styles, 'is-focused')}>
+            {handles[1]}
+          </FocusRing>
+          {upperTrack}</>);
+      }}
+
+    </SliderBase>);
+}
+
+// TODO forwardref?
```

Comment:
Yes. It should be a `DOMRef` pointing at the outer-most element.

Reply:
Should I use `useFocusableRef` or `useDOMRef`?

Reply:
Ah I think `useFocusableRef` with the focusable part pointing to the input makes sense. For range slider, I guess the first input?